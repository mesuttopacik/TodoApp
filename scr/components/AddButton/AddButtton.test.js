import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';
import Addbutton from './AddButton';

test('shoulg match with snapshot', () => {
  const comp = render(<Addbutton />);
  expect(comp).toMatchSnapshot();
});

test('should trigger onPress', () => {
  const mockFunction = jest.fn();
  const comp = render(<Addbutton onClick={mockFunction} />);

  const buttonToucable = comp.getByTestId('button-toucable');
  fireEvent(buttonToucable, 'press');

  expect(mockFunction).toBeCalled();
});
