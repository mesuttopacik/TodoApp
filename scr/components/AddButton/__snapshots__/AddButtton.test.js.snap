// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoulg match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="button-toucable"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D18CE0",
        "borderColor": "#D18CE0",
        "borderRadius": 25,
        "borderWidth": 1,
        "height": 50,
        "justifyContent": "center",
        "width": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFBE9",
          "fontSize": 20,
        }
      }
    >
      +
    </Text>
  </View>
</View>
`;
